packages/client/hmi-client/src/workflow/ops/model-from-document/tera-model-from-document-node.vue

```diff
@@ -1,24 +1,85 @@
 <template>
 	<main>
-		<tera-carousel>
-			<img v-for="(image, index) in images" :src="image" alt="alt" :key="index" />
-		</tera-carousel>
+		<template v-if="model">
+			<SelectButton
+				class="p-button-xsm"
+				:model-value="view"
+				@change="if ($event.value) view = $event.value;"
+				:options="viewOptions"
+			/>
+			<div class="container">
+				<tera-model-diagram
+					v-if="view === ModelNodeView.Diagram"
+					:model="model"
+					:is-editable="false"
+					is-preview
+				/>
+				<tera-model-equation
+					v-else-if="view === ModelNodeView.Equation"
+					:model="model"
+					:is-editable="false"
+				/>
+			</div>
+		</template>
+		<tera-operator-placeholder
+			v-else
+			:operation-type="WorkflowOperationTypes.MODEL_FROM_DOCUMENT"
+		/>
 		<Button @click="emit('open-drilldown')" label="Edit" severity="secondary" outlined />
 	</main>
 </template>
 
 <script setup lang="ts">
-import TeraCarousel from '@/components/widgets/tera-carousel.vue';
-import terarium from '@/assets/svg/terarium-icon.svg';
 import Button from 'primevue/button';
+import { WorkflowNode, WorkflowOperationTypes } from '@/types/workflow';
+import TeraOperatorPlaceholder from '@/components/operator/tera-operator-placeholder.vue';
+import TeraModelEquation from '@/components/model/petrinet/tera-model-equation.vue';
+import TeraModelDiagram from '@/components/model/petrinet/model-diagrams/tera-model-diagram.vue';
+import SelectButton from 'primevue/selectbutton';
+import { ref, onUpdated, onMounted } from 'vue';
+import { ModelOperationState } from '@/workflow/ops/model/model-operation';
+import { getModel } from '@/services/model';
+import { Model } from '@/types/Types';
 
-const emit = defineEmits(['open-drilldown']);
+const props = defineProps<{
+	node: WorkflowNode<ModelOperationState>;
+}>();
+
+enum ModelNodeView {
+	Diagram = 'Diagram',
+	Equation = 'Equation'
+}
+const view = ref(ModelNodeView.Diagram);
+const viewOptions = ref([ModelNodeView.Diagram, ModelNodeView.Equation]);
+
+const model = ref(null as Model | null);
+
+onMounted(
+	async () => {
+		if (props.node?.outputs[0]?.state?.modelId) {
+			model.value = await getModel(props.node.outputs[0].state.modelId);
+		}
+		console.log('model', model.value);
+		console.log('props.node.state.modelId', props.node);
+	},
+	{ immediate: true }
```

Comment:
Not sure what happened here: either you use async and wouldn't have this `{ immediate: true }` thing, or you want to use a watcher syntax.